### PR TITLE
Edits for Postgres compatibility

### LINKED ISSUE #2256: Changes for PostgreSQL compatibility
- PR: https://github.com/cockroachdb/cockroach/pull/19931
  - The `time zone` session variable (with a space) has been renamed `timezone` (without a space), and `SET TIMEZONE` and `SHOW TIMEZONE` are now supported alongside the existing `SET TIME ZONE` and `SHOW TIME ZONE` syntax. Also, `SET TIMEZONE =` can now be used as an alternative to `SET TIMEZONE TO`. 
- PR: https://github.com/cockroachdb/cockroach/pull/19971
  - The `transaction_read_only` session variable is now supported. It is always set to `off`.
- PR: https://github.com/cockroachdb/cockroach/pull/20264
  - The `transaction isolation level`, `transaction priority`, and `transaction status` session variables have been renamed `transaction_isolation`, `transaction_priority`, and `transaction_status`.

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2256, #2175.

### CODE REVIEW

File: v2.0/set-transaction.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ No [privileges](privileges.html) are required to set the transaction isolation l
...
-| `PRIORITY` | If you don't want the transaction to run with `NORMAL` priority, you can set it to `LOW` or `HIGH`.<br/><br/>Transactions with higher priority are less likely to need to be retried.<br/><br/>For more information, see [Transactions: Priorities](transactions.html#transaction-priorities).<br/><br/>**Default**: `NORMAL` |
+| `ISOLATION LEVEL` | By default, transactions in CockroachDB implement the strongest ANSI isolation level: `SERIALIZABLE`. At this isolation level, transactions will never result in anomalies. The `SNAPSHOT` isolation level is still supported as well for backwards compatibility, but you should avoid using it. It provides little benefit in terms of performance and can result in inconsistent state under certain complex workloads. For more information, see [Transactions: Isolation Levels](transactions.html#isolation-levels).<br><br><span class="version-tag">New in v2.0:</span> Alias: `transaction_isolation`<br><br>**Default**: `SERIALIZABLE` |
```
Comment (knz) [inline v2.0/set-transaction.md]: Here and below:

```
<br><br><span class="version-tag">New in v2.0:</span> The current isolation level is also exposed as the [session variable](show-vars.html) `transaction_isolation`.
```

(it's not an alias.)

File: v2.0/set-vars.md [hunk 2]
```diff
@@ -40,8 +40,12 @@ The variable name is case insensitive. The value can be a list of one or more it
...
-| `time zone` | The default time zone for the current session.<br><br>This value can be a string representation of a local system-defined time zone (e.g., `'EST'`, `'America/New_York'`) or a positive or negative numeric offset from UTC (e.g., `-7`, `+7`). Also, `DEFAULT`, `LOCAL`, or `0` sets the session time zone to `UTC`.</br><br>See [`SET TIME ZONE`](#set-time-zone) for more details. | `UTC` | Yes |
+| `server_version_num` | <span class="version-tag">New in v2.0:</span> The version of PostgreSQL that CockroachDB emulates. | Version-dependent | Yes |
+| `time zone` | The default time zone for the current session.<br><br>This value can be a string representation of a local system-defined time zone (e.g., `'EST'`, `'America/New_York'`) or a positive or negative numeric offset from UTC (e.g., `-7`, `+7`). Also, `DEFAULT`, `LOCAL`, or `0` sets the session time zone to `UTC`.</br><br>See [`SET TIME ZONE`](#set-time-zone) for more details. <br><br><span class="version-tag">New in v2.0:</span> Alias: `"timezone"` | `UTC` | Yes |
```
Comment (knz) [inline v2.0/set-vars.md]: The variable has been renamed.

This entire line must be replaced with:

```
| `timezone` | The default time zone for the current session.<br><br>This value can be a string representation of a local system-defined time zone (e.g., `'EST'`, `'America/New_York'`) or a positive or negative numeric offset from UTC (e.g., `-7`, `+7`). Also, `DEFAULT`, `LOCAL`, or `0` sets the session time zone to `UTC`.</br><br>See [Setting the Time Zone](#set-time-zone) for more details. <br><br><span class="version-tag">Changed in v2.0:</span> This session variable was named `"time zone"` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `UTC` | Yes |
```

File: v2.0/set-vars.md [hunk 3]
```diff
@@ -40,8 +40,12 @@ The variable name is case insensitive. The value can be a list of one or more it
...
 | `tracing` | The trace recording state.<br><br>See [`SET TRACING`](#set-tracing) for more details. | `off` | Yes |
+| `transaction isolation level` | The isolation level of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details.<br><br><span class="version-tag">New in v2.0:</span> Alias: `transaction_isolation` | `SERIALIZABLE` | Yes |
```
Comment (knz) [inline v2.0/set-vars.md]: The variables have been renamed.

These three lines must be entirely replaced by:

```
| `transaction_isolation` | The isolation level of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details.<br><br><span class="version-tag">Changed in v2.0:</span> This session variable was called `transaction isolation level` (with spaces) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `SERIALIZABLE` | Yes |
| `transaction_priority` | The priority of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details.<br><br><span class="version-tag">Changed in v2.0:</span> This session variable was called `transaction priority` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `NORMAL` | Yes |
| `transaction_status` | The state of the current transaction. See [Transactions](transactions.html) for more details.<br><br><span class="version-tag">Changed in v2.0:</span> This session variable was called `transaction status` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `NoTxn` | Yes |
```

File: v2.0/set-vars.md [hunk 4]
```diff
@@ -159,7 +163,7 @@ You can control your client's default time zone for the current session with <co
 
 {{site.data.alerts.callout_info}}With setting <code>SET TIME ZONE</code>, CockroachDB uses UTC as the default time zone.{{site.data.alerts.end}}
 
-`SET TIME ZONE` uses a special syntax form used to configure the `"time zone"` session parameter because `SET` cannot assign to parameter names containing spaces.
+`SET TIME ZONE` uses a special syntax form used to configure the `"timezone"` session parameter because `SET` cannot assign to parameter names containing spaces.
```
Comment (knz) [inline v2.0/set-vars.md]: This entire line must be deleted as it is no longer accurate.

File: v2.0/show-vars.md [hunk 5]
```diff
@@ -40,13 +40,15 @@ The variable name is case insensitive. It may be enclosed in double quotes; this
...
 | `server_version` | The version of PostgreSQL that CockroachDB emulates. | Version-dependent | No |
+| `server_version_num` | <span class="version-tag">New in v2.0:</span> The version of PostgreSQL that CockroachDB emulates. | Version-dependent | Yes |
 | `session_user` | The user connected for the current session. | User in connection string | No |
 | `sql_safe_updates` | If `false`, potentially unsafe SQL statements are allowed, including `DROP` of a non-empty database and all dependent objects, `DELETE` without a `WHERE` clause, `UPDATE` without a `WHERE` clause, and `ALTER TABLE .. DROP COLUMN`. See [Allow Potentially Unsafe SQL Statements](use-the-built-in-sql-client.html#allow-potentially-unsafe-sql-statements) for more details. | `true` for interactive sessions from the [built-in SQL client](use-the-built-in-sql-client.html),<br>`false` for sessions from other clients | Yes |
-| `time zone` | The default time zone for the current session   | `UTC` | Yes |
+| `time zone` | The default time zone for the current session. <br><br><span class="version-tag">New in v2.0:</span> Alias: `"timezone"`  | `UTC` | Yes |
```
Comment (knz) [inline v2.0/show-vars.md]: Replace:

```
| `timezone` | The default time zone for the current session. <br><br><span class="version-tag">Changed in v2.0:</span> This session variable was named `"time zone"` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `UTC` | Yes |
```

File: v2.0/show-vars.md [hunk 6]
```diff
@@ -40,13 +40,15 @@ The variable name is case insensitive. It may be enclosed in double quotes; this
...
 | `tracing` | | `off` | |
-| `transaction isolation level` | The isolation level of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details. | `SERIALIZABLE` | Yes |
-| `transaction priority` | The priority of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details. | `NORMAL` | Yes |
-| `transaction status` | The state of the current transaction. See [Transactions](transactions.html) for more details. | `NoTxn` | No |
+| `transaction isolation level` | The isolation level of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details.<br><br><span class="version-tag">New in v2.0:</span> Alias: `transaction_isolation` | `SERIALIZABLE` | Yes |
```
Comment (knz) [inline v2.0/show-vars.md]: Take the 3 lines over from my comment on set-vars.md, adjusting the last column as needed.

File: v2.0/string.md [hunk 7]
```diff
@@ -16,6 +16,7 @@ In CockroachDB, the following are aliases for `STRING`:
 
 - `CHARACTER`
 - `CHAR`
+- `"char"`
```
Comment (knz) [inline v2.0/string.md]: this line is not accurate and should be removed. Just curious: why did you add it?

File: v2.0/set-transaction.md (hunk 1 above)
Comment (knz) [reply v2.0/set-transaction.md]: Also btw this is now new in v2.0.

**Comment by lhirata** [comment]
Review status: 0 of 4 files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[v2.0/set-transaction.md, line 25 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK7--r2-25:-L9pH4S825JQHk8ODM0r:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/set-transaction.md#L25)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Also btw this is now new in v2.0. 
</blockquote></details>

Done.

---

*[v2.0/set-vars.md, line 44 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK70-r2-44:-L9pHhnD47LQw0nvWKQ7:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/set-vars.md#L44)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

The variable has been renamed.

This entire line must be replaced with:

```
| `timezone` | The default time zone for the current session.<br><br>This value can be a string representation of a local system-defined time zone (e.g., `'EST'`, `'America/New_York'`) or a positive or negative numeric offset from UTC (e.g., `-7`, `+7`). Also, `DEFAULT`, `LOCAL`, or `0` sets the session time zone to `UTC`.</br><br>See [Setting the Time Zone](#set-time-zone) for more details. <br><br><span class="version-tag">Changed in v2.0:</span> This session variable was named `"time zone"` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `UTC` | Yes |
```

</blockquote></details>

Done.

---

*[v2.0/set-vars.md, line 46 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK70-r2-46:-L9pMWFr1XliVx3OMpp0:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/set-vars.md#L46)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

The variables have been renamed.

These three lines must be entirely replaced by:

```
| `transaction_isolation` | The isolation level of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details.<br><br><span class="version-tag">Changed in v2.0:</span> This session variable was called `transaction isolation level` (with spaces) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `SERIALIZABLE` | Yes |
| `transaction_priority` | The priority of the current transaction. See [Transaction parameters](transactions.html#transaction-parameters) for more details.<br><br><span class="version-tag">Changed in v2.0:</span> This session variable was called `transaction priority` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `NORMAL` | Yes |
| `transaction_status` | The state of the current transaction. See [Transactions](transactions.html) for more details.<br><br><span class="version-tag">Changed in v2.0:</span> This session variable was called `transaction status` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `NoTxn` | Yes |
```

</blockquote></details>

Done.

---

*[v2.0/set-vars.md, line 158 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9pMlgqEhCZDwaiaqy5:-L9pMlgqEhCZDwaiaqy6:b-5oc28w) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/set-vars.md#L158)):*
> ```Markdown
> ~~~
> 
> ## `SET TIME ZONE`
> ```

@knz - so you can still `SET TIME ZONE`? Or `SET timezone`. Both are valid? 

---

*[v2.0/set-vars.md, line 166 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK70-r2-166:-L9lZWjsBbGbOR2fTz7g:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/set-vars.md#L166)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

This entire line must be deleted as it is no longer accurate.
</blockquote></details>

Done.

---

*[v2.0/show-vars.md, line 46 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK71-r2-46:-L9pN9WD6sfQlbTvTsk7:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/show-vars.md#L46)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Replace:

```
| `timezone` | The default time zone for the current session. <br><br><span class="version-tag">Changed in v2.0:</span> This session variable was named `"time zone"` (with a space) in CockroachDB 1.x. It was renamed for compatibility with PostgreSQL. | `UTC` | Yes |
```
</blockquote></details>

Done.

---

*[v2.0/show-vars.md, line 48 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK71-r2-48:-L9pPSgy88piXtENr11v:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/show-vars.md#L48)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Take the 3 lines over from my comment on set-vars.md, adjusting the last column as needed.
</blockquote></details>

Done.

---

*[v2.0/string.md, line 19 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK72-r2-19:-L9lYdjF-GR2EfYOG1Tf:b-1vrrcr) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/string.md#L19)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

this line is not accurate and should be removed. Just curious: why did you add it?
</blockquote></details>

Removed. Edited per #2175: "The type name "char" (with quotes) is recognized as an alias for CHAR. Need to update https://www.cockroachlabs.com/docs/dev/string.html#aliases."

Does this need to be added somewhere else instead?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2887)*

**Comment by knz** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[v2.0/set-vars.md, line 158 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9pMlgqEhCZDwaiaqy5:-L9pWCbh-4SMxu-uQB4K:b-acxfsq) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/set-vars.md#L158)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

@knz - so you can still `SET TIME ZONE`? Or `SET timezone`. Both are valid? 
</blockquote></details>

Yes. That's what the table "Special syntax cases" above explains.

---

*[v2.0/set-vars.md, line 61 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9pWKZW-Mz4A7g9BdIz:-L9pWKZW-Mz4A7g9BdJ-:beh43xj) ([raw file](https://github.com/cockroachdb/docs/blob/e2e77f2bca82761e40c7512ea362c73125f98116/v2.0/set-vars.md#L61)):*
> ```Markdown
> | `SET NAMES ...` | `SET client_encoding = ...` | This is provided for compatibility with PostgreSQL clients.
> | `SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL ...` | `SET default_transaction_isolation = ...` | This is provided for compatibility with standard SQL.
> | `SET TIME ZONE ...` | `SET "timezone" = ...` | This is provided for compatibility with PostgreSQL clients.
> ```

You can change `SET "timezone" = ` to `SET timezone = `

---

*[v2.0/string.md, line 19 at r2](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK72-r2-19:-L9pW_uWC3qTLd68Nr_J:b-h7wyhq) ([raw file](https://github.com/cockroachdb/docs/blob/c619dcf1eb0dfc0a7c5c860c787af03d7fa3a8b0/v2.0/string.md#L19)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

Removed. Edited per #2175: "The type name "char" (with quotes) is recognized as an alias for CHAR. Need to update https://www.cockroachlabs.com/docs/dev/string.html#aliases."

Does this need to be added somewhere else instead?
</blockquote></details>

Thanks for bringing my attention to this. The code has a bug. I filed https://github.com/cockroachdb/cockroach/issues/24686 to track this. Let's not document this until the code becomes correct.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-:-L9pWlIM4a5RYCbxRup_:bnfp4nl)*

**Comment by lhirata** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[v2.0/set-vars.md, line 61 at r3](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9pWKZW-Mz4A7g9BdIz:-L9pbDB4EhMGr5rt5iVF:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/e2e77f2bca82761e40c7512ea362c73125f98116/v2.0/set-vars.md#L61)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

You can change `SET "timezone" = ` to `SET timezone = `
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2887)*

**Comment by lhirata** [comment]
Review status: 3 of 4 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[v2.0/set-vars.md, line 48 at r4](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK70-r4-48:-LAOcneV1vsf94IyJ9kT:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/441e4c7e92399dfba61fdcc519febec76efa9156/v2.0/set-vars.md#L48)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

I would change all uses of "It was renamed ..." to "It has been renamed ..."

Reasoning: The sentence just before says "This ... variable was called X", which denotes a previous state, so IMO saying "It was" again recalls that previous state, whereas saying "it has been" makes clear that the variable name moved to a new state.
</blockquote></details>

Done.

---

*[v2.0/show-vars.md, line 48 at r4](https://beta.reviewable.io/reviews/cockroachdb/docs/2887#-L9flHfr_6OWIIV8FK71-r4-48:-LAOcpYd57NXPOKfyRUK:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/441e4c7e92399dfba61fdcc519febec76efa9156/v2.0/show-vars.md#L48)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Re: `timezone` and `transaction_*`, same comment/suggestion as above re: using "has been renamed for compatibility" in the closing sentence.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/docs/2887)*
